src/lib/routers/history.ts

```diff
@@ -0,0 +1,197 @@
+import qs from 'qs';
+import { RouteState } from '../../types';
+
+type CreateUrlOptions = {
```

Comment:
IIRC we use a lot `URL` vs `Url` inside the codebase. We should stay consistent.